### PR TITLE
Fix build warnings

### PR DESCRIPTION
**Pull request by ValarDragon**
## What is the purpose of the change

Fix build warnings

## Brief Changelog

* Deletes invalid [nullable=false] on repeated fields for gamm stableswap. (This is what keeps giving warnings on make proto all)
* Deletes the currently unused `sims.mk`, cc @czarcas7ic if you think it makes sense to re-instate, but we'd do that in a separate PR. (and fix the build warning)

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: proto/osmosis/gamm/pool-models/stableswap/stableswap_pool.proto [hunk 1]
```diff
@@ -64,10 +64,8 @@ message Pool {
     (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
   ];
   // for calculation amognst assets with different precisions
-  repeated uint64 scaling_factor = 7 [
-    (gogoproto.moretags) = "yaml:\"stableswap_scaling_factor\"",
-    (gogoproto.nullable) = false
```
Comment (mattverse) [inline proto/osmosis/gamm/pool-models/stableswap/stableswap_pool.proto:69]: I've testing comparing build warnings with this branch vs main locally, the only build warnings this branch suppressed was `Makefile:242: warning: overriding commands for target test-sim-benchmark
sims.mk:61: warning: ignoring old commands for target test-sim-benchmark` which seems to come from sims.mk.

I was curious why we were removing these lines? (Maybe something in my local env that doesn't trigger warnings related to this?)

File: Makefile [hunk 2]
```diff
@@ -214,8 +214,6 @@ PACKAGES_E2E=$(shell go list -tags e2e ./... | grep '/e2e')
 PACKAGES_SIM=$(shell go list ./... | grep '/tests/simulator')
 TEST_PACKAGES=./...
 
-include sims.mk
```
Comment (mattverse) [inline Makefile:217]: asking to learn: was this never used by anyone?

**Review by mattverse — COMMENTED** [review]
Removing sims.mk lgtm for suppressing build warnings! Left some comments and questions

File: proto/osmosis/gamm/pool-models/stableswap/stableswap_pool.proto (hunk 1 above)
Comment (ValarDragon) [reply proto/osmosis/gamm/pool-models/stableswap/stableswap_pool.proto:69]: `make proto-all` does this

File: Makefile (hunk 2 above)
Comment (ValarDragon) [reply Makefile:217]: We have never used this in Osmosis

File: proto/osmosis/gamm/pool-models/stableswap/stableswap_pool.proto (hunk 1 above)
Comment (mattverse) [reply proto/osmosis/gamm/pool-models/stableswap/stableswap_pool.proto:69]: ah gotcha

**Review by mattverse — APPROVED** [review]

**Pull request merged**